client/hwlib/src/constants.rs

```diff
@@ -23,6 +23,9 @@ pub const PROC_CPUINFO_FILE_PATH: &str = "/proc/cpuinfo";
 pub const PROC_DEVICE_TREE_DIR_PATH: &str = "/proc/device-tree/";
 pub const PROC_VERSION_FILE_PATH: &str = "/proc/version";
 
+// The os-release file path is set during the build time
+// to allow for snap packaging, which exposes the file in a different location.
+pub const OS_RELEASE_FILE_PATH: &str = env!("OS_RELEASE_FILE_PATH", "/etc/os-release");
```

Comment:
Is there a little code-duplication here? I think the fallback to `/etc/os-release` is already handled by `build.rs`---if something changes and `build.rs` no longer defines the env var we expect, I think it could be nice to trigger a compilation error here (i.e could the fallback on this line be removed?)